compose/foundation/foundation/src/skikoMain/kotlin/androidx/compose/foundation/OnClick.skiko.kt

```diff
@@ -0,0 +1,444 @@
+                            delayPressInteraction = mutableStateOf({ false })
+                        )
+                    }
+                )
+            }
+        } else {
+            Modifier
+        }
+
+        gestureModifier
+    }
+)
+
+private val EmptyPointerEvent = PointerEvent(emptyList())
+
+/**
+ * Detects tap, double-tap, and long press gestures and calls [onTap], [onDoubleTap], and
+ * [onLongPress], respectively, when detected. [onPress] is called when the press is detected
+ * and the [PressGestureScope.tryAwaitRelease] and [PressGestureScope.awaitRelease] can be
+ * used to detect when pointers have released or the gesture was canceled.
+ * The first pointer down and final pointer up are consumed, and in the
+ * case of long press, all changes after the long press is detected are consumed.
+ *
+ * When [onDoubleTap] is provided, the tap gesture is detected only after
+ * the [ViewConfiguration.doubleTapMinTimeMillis] has passed and [onDoubleTap] is called if the
+ * second tap is started before [ViewConfiguration.doubleTapTimeoutMillis]. If [onDoubleTap] is not
+ * provided, then [onTap] is called when the pointer up has been received.
+ *
+ * If the first down event was consumed, the entire gesture will be skipped, including
+ * [onPress]. If the first down event was not consumed, if any other gesture consumes the down or
+ * up events, the pointer moves out of the input area, or the position change is consumed,
+ * the gestures are considered canceled. [onDoubleTap], [onLongPress], and [onTap] will not be
+ * called after a gesture has been canceled.
+ *
+ * @param matcher defines supported pointer types and required properties
+ * @param keyboardModifiers defines a condition that [PointerEvent.keyboardModifiers] has to match
+ */
+@ExperimentalFoundationApi
+suspend fun PointerInputScope.detectTapGestures(
+    matcher: PointerMatcher = PointerMatcher.Primary,
+    keyboardModifiers: PointerKeyboardModifiers.() -> Boolean = { true },
+    onDoubleTap: ((Offset) -> Unit)? = null,
+    onLongPress: ((Offset) -> Unit)? = null,
+    onPress: suspend PressGestureScope.(Offset) -> Unit = { },
+    onTap: ((Offset) -> Unit)? = null
+) = coroutineScope {
+    // special signal to indicate to the sending side that it shouldn't intercept and consume
+    // cancel/up events as we're only require down events
+    val pressScope = PressGestureScopeImpl(this@detectTapGestures)
+
+    val filter: (PointerEvent) -> Boolean = {
+        matcher.matches(it) && keyboardModifiers(it.keyboardModifiers)
+    }
+
+    while (currentCoroutineContext().isActive) {
+        awaitPointerEventScope {
+            pressScope.reset()
+            while (
+                currentEvent != EmptyPointerEvent &&
+                currentEvent.isAllPressedDown(false) &&
+                matcher.matches(currentEvent)
+            ) {
+                // suspend until currentEvent matches the filter for pressed event
```

Comment:
Do we need this code? When I debugged it, `isAllPressedDown` was always false

Reply:
It was necessary when we relied on PointerButtons.
But now, we use PointerButton, so it's not needed.